aiohttp/client.py

```diff
@@ -576,7 +582,19 @@ async def _request(
 
                     # Override the auth with the one from the URL only if we
                     # have no auth, or if we got an auth from a redirect URL
-                    if auth is None or (history and auth_from_url is not None):
+                    if (auth is None or history) and auth_from_url is not None:
+                        # Pre-check the credentials can be encoded to
+                        # avoid crashing down the line
+                        try:
+                            auth_from_url.encode()
+                        except UnicodeEncodeError as e:
```

Comment:
This looks like a breaking change for anyone trapping `UnicodeEncodeError` currently since the new exceptions don't include it in the inheritance.

Reply:
We could avoid the breaking change by making the new exceptions also derived from `UnicodeEncodeError`

Reply:
Good point, but hypothetically a case could be added later where these exceptions no longer stem from a `UnicodeEncodeError` so that would also make it weird.

If we don't want to make the exceptions as generic as possible now, I'd include `UnicodeEncodeError` in the exception names in addition to the inheritance to avoid someone reusing it for some other root error down the line.

Reply:
Maybe we could use `InvalidRedirectUrlAuthClientUnicodeEncodeError` and `InvalidUrlAuthClientUnicodeEncodeError`, which inherit from both.  
I know, they're a bit of a mouthful 😉

or maybe just `InvalidRedirectUrlAuthClientUnicodeError` and `InvalidUrlAuthClientUnicodeError`

Reply:
What about a shorter `[Redirect]UrlAuthUnicodeEncodingError`?